app/src/org/commcare/engine/cases/AndroidCaseInstanceTreeElement.java

```diff
@@ -91,8 +95,9 @@ protected synchronized void loadElements() {
         String[] namesToMatch = new String[numKeys];
         String[] valuesToMatch = new String[numKeys];
         for (int i = numKeys - 1; i >= 0; i--) {
-            namesToMatch[i] = keys.elementAt(i);
-            valuesToMatch[i] = (String)values.elementAt(i);
+            namesToMatch[i] = profiles.elementAt(i).getKey();
+            valuesToMatch[i] = (String)

```

Comment:
Why do we know this to be a String? OR why does IndexedValueLookup allow the ambiguity of an Object?